test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/RuntimeTypeInfoTest.cs

```diff
@@ -0,0 +1,403 @@
+                }
+
+                set
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public int Count
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public bool IsReadOnly
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public ICollection<List<string>> Keys
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public ICollection<string> Values
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public void Add(KeyValuePair<List<string>, string> item)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void Add(List<string> key, string value)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void Clear()
+            {
+                throw new NotImplementedException();
+            }
+
+            public bool Contains(KeyValuePair<List<string>, string> item)
+            {
+                throw new NotImplementedException();
+            }
+
+            public bool ContainsKey(List<string> key)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void CopyTo(KeyValuePair<List<string>, string>[] array, int arrayIndex)
+            {
+                throw new NotImplementedException();
+            }
+
+            public IEnumerator<KeyValuePair<List<string>, string>> GetEnumerator()
+            {
+                throw new NotImplementedException();
+            }
+
+            public bool Remove(KeyValuePair<List<string>, string> item)
+            {
+                throw new NotImplementedException();
+            }
+
+            public bool Remove(List<string> key)
+            {
+                throw new NotImplementedException();
+            }
+
+            public bool TryGetValue(List<string> key, out string value)
+            {
+                throw new NotImplementedException();
+            }
+
+            IEnumerator<int> IEnumerable<int>.GetEnumerator()
+            {
+                throw new NotImplementedException();
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public class KnownKeyDictionary<TValue> : Dictionary<string, TValue>
+        {
+        }
+
+        private class CustomType
+        {
+        }
+    }
+}
+
+namespace Microsoft.AspNet.Razor.Fake
```

Comment:
Move to separate files. Everything here is at the top level and visible throughout the project. So there's no upside to burying the definitions here.